pkg/vulnerability/vulnerability.go

```diff
@@ -120,7 +131,45 @@ func (c Client) Filter(vulns []types.DetectedVulnerability, severities []dbTypes
 		}
 		return vulnerabilities[i].VulnerabilityID < vulnerabilities[j].VulnerabilityID
 	})
-	return vulnerabilities
+	return vulnerabilities, nil
+}
+
+func applyPolicy(ctx context.Context, vulns []types.DetectedVulnerability, policyFile string) ([]types.DetectedVulnerability, error) {
+	policy, err := ioutil.ReadFile(policyFile)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to read the policy file: %w", err)
+	}
+
+	query, err := rego.New(
+		rego.Query("data.trivy.allow"),
+		rego.Module("lib.rego", module),
+		rego.Module("trivy.rego", string(policy)),
+	).PrepareForEval(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to prepare for eval: %w", err)
+	}
+
+	var filtered []types.DetectedVulnerability
+	for _, vuln := range vulns {
+		results, err := query.Eval(ctx, rego.EvalInput(vuln))
+		if err != nil {
+			return nil, xerrors.Errorf("unable to evaluate the policy: %w", err)
+		} else if len(results) == 0 {
+			// Handle undefined result.
+			filtered = append(filtered, vuln)
+			continue
+		}
+		allow, ok := results[0].Expressions[0].Value.(bool)
```

Comment:
If the user doesn't explicitly default the allow rule to be false, this will fail when indexing into the `Expressions`. My other comment suggests adding documentation about this but it's better to not rely on user input and check here as well.

Reply:
If `allow` is undefined, I think `results` should be empty and it is handled here. I removed `default allow=false` and confirmed that it returned empty `results`.
https://github.com/aquasecurity/trivy/blob/d3df71c6378e3bfe6c096a64562bfde91462298f/pkg/vulnerability/vulnerability.go#L158-L160

I couldn't find when `Expresssions` would be empty.
https://github.com/open-policy-agent/opa/blob/v0.22.0/rego/rego.go#L1698-L1700